ECP: Add Unit Tests for secp224k1

minosgalanakis opened the pull request:
## Description

Resolves #7257 


## Gatekeeper checklist

- [x] **changelog**not required. Extending testing coverage, does not affect library's behaviour
- [x] **backport** not required. This is worked required by a newly developped interface
- [x] **tests** provided

gabor-mezei-arm commented on tests/suites/test_suite_ecp.function:
```diff
@@ -1355,6 +1355,89 @@ exit:
+
+    TEST_EQUAL(mbedtls_ecp_mod_p224k1(&X), 0);
+    TEST_LE_U(mbedtls_mpi_core_bitlen(X.p, X.n), 224);
+    mbedtls_mpi_mod_raw_fix_quasi_reduction(X.p, &m);
```
As the description of this issue recommend  the `mbedtls_mpi_mod_mpi()` should be used here.

gabor-mezei-arm reviewed (changes requested):
The changes for the P192K1 testing is not part of this PR and must be removed.

yanesca commented:
@minosgalanakis this is a deliverable for this quarter, it should have a `priority-high` label, could you please add it?

minosgalanakis commented:
I have rebased and addressed the comment. The code is still based on #7424 since it makes resolving conflicts easier and we can test the code as we expect it to come in.

Please only review the [last three commits](https://github.com/Mbed-TLS/mbedtls/pull/7425/files/cec506083b3dbddd52bdf6d22d5b18c4277f5aae..56ca0e3c50ab498eacb9a2c79d1897a5268a8778)

When the pr for `p192k1` is merged, I will rebase it

paul-elliott-arm reviewed (approved):
LGTM
